libraries/lib-project-history/UndoManager.cpp

```diff
@@ -228,11 +226,6 @@ void UndoManager::PushState(const TranslatableString &longDescription,
        lastAction.Translation() == longDescription.Translation() &&
        mayConsolidate ) {
       ModifyState();
-      // MB: If the "saved" state was modified by ModifyState, reset
-      //  it so that UnsavedChanges returns true.
-      if (current == saved) {
-         saved = -1;
-      }
       return;
```

Comment:
I believe we need a call to MarkUnsaved() here

Reply:
I think you're right. The comment mislead me (`ModifyState()` in fact never changed `saved`), yet is a new state is pushed, of course there is something to save.